### PR TITLE
get resume id from server

### PR DESCRIPTION
**Pull request by coderlxn**

### CODE REVIEW

**Comment by olesalscheider** [comment]
LGTM.

**Comment by jlaine** [comment]
IQ parsing / serialization is trivial to unit test, so do it :)

**Comment by jlaine** [comment]
@coderlxn can we please have a unit test for this so we can merge it?

**Comment by jlaine** [comment]
@lnjX I'm not sure I understand what this does anymore, ideas?

File: src/base/QXmppStreamManagement.cpp [hunk 1]
```diff
@@ -132,6 +132,7 @@ void QXmppStreamManagementEnabled::parse(const QDomElement &element)
     m_resume = resume == QString("true") || resume == QString("1");
     m_max = element.attribute("max").toUInt();
     m_location = element.attribute("location");
+    m_id = element.attribute("id");
```
Comment (lnjX) [inline src/base/QXmppStreamManagement.cpp]: This only adds parsing, we also need to add this in `::toXml`.

**Comment by lnjX** [comment]
@jlaine The session id needs to be parsed to be able to resume the session (using `previd` in `<resume/>`) after a possible connection loss.

**Comment by lnjX** [comment]
Unfortunately there's no test class for stream management, yet, but you could base on [this test](https://github.com/qxmpp-project/qxmpp/blob/master/tests/qxmppiq/tst_qxmppiq.cpp) basically. It were also nice to have this new test class in a new commit.

**Comment by lnjX** [comment]
Parsing of the id attribute has been added in the meantime, so this is obsolete.

**Pull request closed without merging**